google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/PolicyHelper.java

```diff
@@ -33,10 +33,15 @@
 
   static Policy convertFromApiPolicy(com.google.api.services.storage.model.Policy apiPolicy) {
     Policy.Builder policyBuilder = Policy.newBuilder();
-    for (Bindings binding : apiPolicy.getBindings()) {
-      for (String member : binding.getMembers()) {
-        policyBuilder.addIdentity(Role.of(binding.getRole()), Identity.valueOf(member));
+    List<Bindings> bindings = apiPolicy.getBindings();
+    if (null != bindings && bindings.size() > 0) {
+      for (Bindings binding : bindings) {
+        for (String member : binding.getMembers()) {
+          policyBuilder.addIdentity(Role.of(binding.getRole()), Identity.valueOf(member));
+        }
       }
+    } else {
+      throw new IllegalStateException("Missing required bindings.");
```

Comment:
How did you decided that you have to throw IllegalStateException if `apiPolicy.getBindings()` is returning null?

Reply:
I have checked in other products if something is missing then they will directly throw an IllegalStateException based on that i have decided to throw that exception.

Reply:
Ok